worlds/seaofthieves/Regions/Regions.py

```diff
@@ -0,0 +1,88 @@
+import copy
+import typing
+
+from BaseClasses import MultiWorld, Region, Entrance
+from worlds.seaofthieves.Locations.Locations import *
+from ..Locations.LocationCollection import LocationDetailsCollection
+from ...generic.Rules import add_rule, exclusion_rules
+from ..Items.Items import Items
+from ..Configurations import SotOptionsDerived
+import typing
+from worlds.seaofthieves.Regions.RegionDetails import RegionDetails, Regions
+from worlds.seaofthieves.Regions.ConnectionDetails import ConnectionDetails
+
+
+class SOTRegion(Region):
+    subregions: typing.List[Region] = []
+
+    def __init__(self, regionName: str, player: int, world, hint: typing.Optional[str] = None):
+        super().__init__(regionName, player, world, hint)
+
+
+class RegionAdder:
+
+    def __init__(self, player: int, locationDetailsCollection: LocationDetailsCollection,
+                 options: SotOptionsDerived.SotOptionsDerived):
+        self.player = player
+
+        self.locationDetailsCollection = locationDetailsCollection
+        self.options: SotOptionsDerived.SotOptionsDerived = options
+
+        self.__added_regions: typing.List[SOTRegion] = list()
+        pass
+
+    def addRulesForLocationsInRegions(self, world: MultiWorld):
```

Comment:
```suggestion
    def addRulesForLocationsInRegions(self, multiworld: MultiWorld):
```
Do not use `world: MultiWorld`